pandas/tests/indexes/test_base.py

```diff
@@ -1047,6 +1047,26 @@ def test_setops_disallow_true(self, method):
         with pytest.raises(ValueError, match="The 'sort' keyword only takes"):
             getattr(idx1, method)(idx2, sort=True)
 
+    def test_setops_preserve_object_dtype(self):
```

Comment:
k, we should try to split this out the setops to a separate test file (IIRC there is another PR doing of some of this)

Reply:
agreed.  i have a local branch separating some of this into tests/indexes/object/... will push it before long